Change BCF-XML import to import by default.

wielinde opened the pull request:
Add i18n to import wizzard.

https://community.openproject.com/wp/30635

wielinde committed:
Change BCF-XML import to import by default.

Add i18n to import wizzard.

HDinger commented on modules/bcf/app/views/bcf/issues/configure_unknown_statuses.html.erb:
```diff
@@ -3,23 +3,15 @@
+    <input type="hidden" name="import_options[unknown_statuses_action]" value="chose">
+    <%= t('bcf.bcf_xml.import.import_statuses_as') %>
     <% options = Status.all.collect { |status| [status.name, status.id] } %>
     <%= select_tag 'import_options[unknown_statuses_chose_ids][]', options_for_select(options),
```
Please do not use `p` tags for something that is no paragraph and also no unstyled select fields and labels. We have a ruby helper tag `styled_select_tag` which can be used for the correct classes. Otherwise for example mobile styles won't be applied and changes made to our fields won't be applied later.
So it should look somehow like this: 
```
<div class="form--field">
       <%= styled_label_tag 'import_options[unknown_types_chose_ids][],  t('bcf.bcf_xml.import.import_types_as') %> %>
        <%= styled_select_tag('import_options[unknown_types_chose_ids][],
                             options,
                              container_class: '-slim') %>
      </div>
```

The same applies for the other view files.

wielinde replied:
Thank you. The `<P>` tags "smelled" like bad style, when I was writing them. I will also try to use the `styled_select_tag` helper.

wielinde replied:
I just pushed some changes that use notifcation boxes and form fields instead of plain P tags. However, I could not really see real changes to the rendering of the select box. Should I see changes?

Apart from that I had to introduce some ugly NBSPs and BRs for padding/margin. I bet there is a nicer way. Inline styles? Are ugly, too.

HDinger replied:
The select field itself looks pretty much the same. So there is no big difference to see. 

The `nbsp` and `br` however are not necessary. You missed some classes, that is why it did not add the margin by itself. In detail, the label should be really a `label`-tag with the according class `form--label` and a `for` attribute (for accessibility reasons). Further the container `form--field-container` is missing. 
The resulting HTML should look like this:

```
<div class="form--field">
    <input type="hidden" name="import_options[unknown_types_action]" value="chose">
    <label class="form--label" for="import_options[unknown_types_chose_ids][]">Set all these types to</label>
    <div class="form--field-container">
      <span class=" form--select-container -slim">
        <select name="import_options[unknown_types_chose_ids][]" id="unknown_types_chose_ids" class="form--select form--inline-select">
           <option value="6">Bug</option>
           <option value="2">Task</option>
           <option value="1">none</option>
        </select>
     </span>
  </div>
</div>
```

HDinger commented on modules/bcf/config/locales/en.yml:
```diff
@@ -28,9 +31,29 @@ en:
+        invalid_emails_found: 'Invalid email addresses found'
+        unknown_emails_found: 'Unknown email addresses found'
+        non_members_found: 'Non project members found'
+        import_types_as: 'Import them all setting type to'
```
This formulation sounds very strange to me. I had a look a translator and seems correct. However maybe we could simple use `Import all types as`. What do you think?

wielinde replied:
Hmm, I agree that the current string look awkward. What I want to communicate is

- You overwrite those that are listed and not the others. Therefore I use the word "them".
- I wanted to repeat the word "type" in the sentence as this is probably the only line of text that you read when you have already imported several times and do not read the heading anymore.

For now I change it to "Set all these types to". If you or others dislike it, we can change it anytime.

HDinger replied:
👍 That is fine for me

HDinger reviewed (changes requested):
Hi @wielinde 

thanks for the effort with the translations. 👍 

I have a remark regarding the HTML structure, to match our style guide. Please apply the classes and containers so that will be easier to maintain in the future. Feel free to ask me if you're not sure about the correct structure. In general you can always look at `http://localhost:3000/assets/styleguide.html` if you are unsure about the way an (HTML)-element should be defined.

Otherwise it works fine. The import is now much more easy 👍

wielinde committed:
Use styleguide conform block elements and select helpers

Use notifacation boxes and form fields instead of Ps. Also improve wording within the wizzard.

HDinger commented on modules/bcf/app/views/bcf/issues/configure_non_members.html.erb:
```diff
@@ -3,54 +3,48 @@
+                              tabIndex: 0,
+                              class:    'form--select form--inline-select',
+                              container_class: '-slim',
+                              id:       'non_members_add_role_ids' %>
```
You do not need to add the class `form--select` manually, the `styled_select_tag` takes care of this. Further the styles of the modifier `-slim` is overruled by the `form--inline-select` class. So one of the can be removed.

HDinger committed:
Extract reused code to partials and apply correct html structure & classes

HDinger reviewed (approved):
Waiting to travis, afterwards it is good to merge 👍
